contracts/auction/src/contract.rs

```diff
@@ -454,64 +291,87 @@ fn is_deposit_open(current_timestamp: u64, config: &Config) -> bool {
-pub fn handle_withdraw_ust(
+pub fn execute_withdraw(
     deps: DepsMut,
     env: Env,
     info: MessageInfo,
-    amount: Uint128,
+    amount_usdc: Uint128,
+    amount_atom: Uint128,
 ) -> Result<Response, StdError> {
     let config = CONFIG.load(deps.storage)?;
     let mut state = STATE.load(deps.storage)?;
-
     let user_address = info.sender;
-
-    let mut user_info = USERS.load(deps.storage, &user_address)?;
+    let users_store = get_users_store();
+    let mut user_info = users_store.load(deps.storage, &user_address)?;
 
     // CHECK :: Has the user already withdrawn during the current window
-    if user_info.ust_withdrawn {
+    if user_info.withdrawn {
         return Err(StdError::generic_err("Max 1 withdrawal allowed"));
     }
 
     // Check :: Amount should be within the allowed withdrawal limit bounds
     let max_withdrawal_percent = allowed_withdrawal_percent(env.block.time.seconds(), &config);
-    let max_withdrawal_allowed = user_info.ust_delegated * max_withdrawal_percent;
+    let max_allowed_usdc = user_info.usdc_deposited * max_withdrawal_percent;
+    let max_allowed_atom = user_info.atom_deposited * max_withdrawal_percent;
 
-    if amount > max_withdrawal_allowed {
+    if amount_usdc > max_allowed_usdc || amount_atom > max_allowed_atom {
         return Err(StdError::generic_err(format!(
             "Amount exceeds maximum allowed withdrawal limit of {}",
             max_withdrawal_percent
         )));
     }
 
+    if amount_usdc.gt(&Uint128::zero()) && amount_atom.gt(&Uint128::zero()) {
+        return Err(StdError::generic_err(
+            "At least one token must be withdrawn",
+        ));
+    }
+
     // After deposit window is closed, we allow to withdraw only once
     if env.block.time.seconds() >= config.init_timestamp + config.deposit_window {
-        user_info.ust_withdrawn = true;
+        user_info.withdrawn = true;
+    }
+
+    let mut res = Response::new();
+
+    if amount_usdc.gt(&Uint128::zero()) {
+        // Transfer Native tokens to the user
+        let transfer_msg = CosmosMsg::Bank(BankMsg::Send {
+            to_address: user_address.to_string(),
+            amount: vec![Coin {
+                denom: config.usdc_denom,
+                amount: amount_usdc,
+            }],
+        });
+        res = res.add_message(transfer_msg);
+    }
+
+    if amount_atom.gt(&Uint128::zero()) {
+        // Transfer Native tokens to the user
+        let transfer_msg = CosmosMsg::Bank(BankMsg::Send {
+            to_address: user_address.to_string(),
+            amount: vec![Coin {
+                denom: config.atom_denom,
+                amount: amount_atom,
+            }],
+        });
+        res = res.add_message(transfer_msg);
     }
 
     // UPDATE STATE
-    state.total_ust_delegated -= amount;
-    user_info.ust_delegated -= amount;
+    state.total_usdc_deposited -= amount_usdc;
+    state.total_atom_deposited -= amount_atom;
```

Comment:
I don't see where you update `user_info.usdc_deposited` or `user_info.atom_deposited`

Reply:
https://github.com/neutron-org/neutron-tge-contracts/blob/08cfea21db6356ce992b1e2ab87303446917d7b9/contracts/auction/src/contract.rs#L184

Reply:
In the deposit handler I see it for sure, but not in the withdraw handler

Reply:
fixed